extension_service/datastore/providers/firestore.py

```diff
@@ -0,0 +1,270 @@
+                    "departure_time": flight.departure_time,
+                    "arrival_time": flight.arrival_time,
+                    "departure_gate": flight.departure_gate,
+                    "arrival_gate": flight.arrival_gate,
+                }
+            )
+
+    async def export_data(
+        self,
+    ) -> tuple[list[models.Airport], list[models.Amenity], list[models.Flight]]:
+        airport_docs = self.__client.collection("airports").stream()
+        amenities_docs = self.__client.collection("amenities").stream()
+        flights_docs = self.__client.collection("flights").stream()
+
+        airports, amenities, flights = [], [], []
+
+        for doc in airport_docs:
+            airport_dict = doc.to_dict()
+            airport_dict["id"] = doc.id
+            airports.append(models.Airport.model_validate(airport_dict))
+
+        for doc in amenities_docs:
+            amenity_dict = doc.to_dict()
+            amenity_dict["id"] = doc.id
+            amenities.append(models.Amenity.model_validate(amenity_dict))
+
+        for doc in flights_docs:
+            flight_dict = doc.to_dict()
+            flight_dict["id"] = doc.id
+            flights.append(models.Flight.model_validate(flight_dict))
+
+        return airports, amenities, flights
+
+    async def get_airport_by_id(self, id: int) -> Optional[models.Airport]:
+        query = self.__client.collection("airports").where(
+            filter=FieldFilter("id", "==", id)
+        )
+        return models.Airport.model_validate(query.stream().to_dict())
+
+    async def get_airport_by_iata(self, iata: str) -> Optional[models.Airport]:
+        query = self.__client.collection("airports").where(
+            filter=FieldFilter("iata", "==", iata)
+        )
+        return models.Airport.model_validate(query.stream().to_dict())
+
+    async def search_airports(
+        self,
+        country: Optional[str] = None,
+        city: Optional[str] = None,
+        name: Optional[str] = None,
+    ) -> list[models.Airport]:
+        query = self.__client.collection("airports")
+
+        if country:
+            query = query.where("country", "==", country)
+
+        if city:
+            query = query.where("city", "==", city)
+
+        if name:
+            query = query.where("name", ">=", name).where("name", "<=", name + "\uf8ff")
+
+        docs = await query.stream()
+
+        airports = [models.Airport.model_validate(dict(doc)) for doc in docs]
+        return airports
+
+    async def get_amenity(self, id: int) -> Optional[models.Amenity]:
+        query = (
+            self.__client.collection("amenities")
+            .where(filter=FieldFilter("id", "==", id))
+            .select(
+                "id",
+                "name",
+                "description",
+                "location",
+                "terminal",
+                "category",
+                "hour",
+                "content",
+                "embedding",
+            )
```

Comment:
Do we need to specify select here? Not sure we do it in other places

Reply:
Previously `amenities_search` in postgres.py has selection clause so I matched its query. Seems like it is no longer the case so I will delete select clause here as well.

Reply:
I don't see anything missing, so I'm guessing not.